### PR TITLE
Update free-programming-books-fr.md

### PR DESCRIPTION
**Pull request by cuitlazotlac**
hey ! just adding minor changes in french translations

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
changes break links in index. please change index as well - your translation is better

**Comment by cuitlazotlac** [comment]
hey @eshellman - sorry forgot to modify that, it's done now :) let me know if I can help somewhere else

File: books/free-programming-books-fr.md [hunk 1]
```diff
@@ -3,10 +3,10 @@
 * [0 - Méta-listes](#0---méta-listes)
 * [1 - Non dépendant du langage](#1---non-dépendant-du-langage)
     * [Algorithmique](#algorithmique)
-    * [IDE and editors](#ide-and-editors)
+    * [IDE et éditeurs de texte](#ide-and-editors)
     * [Logiciels libres](#logiciels-libres)
     * [Makefile](#makefile)
-    * [Pédagogie pour enfants et adolescents](#pédagogie-pour-enfants-et-adolescents)
+    * [Pédagogie pour les enfants et adolescents](#pédagogie-pour-enfants-et-adolescents)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:9]:
```suggestion
    * [Pédagogie pour les enfants et adolescents](#pédagogie-pour-les-enfants-et-adolescents)
```

File: books/free-programming-books-fr.md [hunk 2]
```diff
@@ -3,10 +3,10 @@
 * [0 - Méta-listes](#0---méta-listes)
 * [1 - Non dépendant du langage](#1---non-dépendant-du-langage)
     * [Algorithmique](#algorithmique)
-    * [IDE and editors](#ide-and-editors)
+    * [IDE et éditeurs de texte](#ide-and-editors)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:6]:
```suggestion
    * [IDE et éditeurs de texte](#ide-et-editeurs-de-texte)
```

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**